modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/MuleExtensionUtils.java

```diff
@@ -0,0 +1,230 @@
+                                  Joiner.on(", ").join(clashes))
+            );
+        }
+    }
+
+    private static Set<String> collectRepeatedNames(Collection<? extends Described> describedCollection, String describedEntityName)
+    {
+        if (CollectionUtils.isEmpty(describedCollection))
+        {
+            return ImmutableSet.of();
+        }
+
+        Multiset<String> names = LinkedHashMultiset.create();
+
+        for (Described described : describedCollection)
+        {
+            checkArgument(described != null, String.format("A null %s was provided", describedEntityName));
+            names.add(described.getName());
+        }
+
+        names = Multisets.copyHighestCountFirst(names);
+        Set<String> repeatedNames = new HashSet<>();
+        for (String name : names)
+        {
+            if (names.count(name) == 1)
+            {
+                break;
+            }
+
+            repeatedNames.add(name);
+        }
+
+        return repeatedNames;
+    }
+
+    public static <T> List<T> immutableList(Collection<T> collection)
+    {
+        return collection != null ? ImmutableList.copyOf(collection) : ImmutableList.<T>of();
+    }
+
+    public static <T extends Described> Map<String, T> toMap(List<T> objects)
+    {
+        ImmutableMap.Builder<String, T> map = ImmutableMap.builder();
+        for (T object : objects)
+        {
+            map.put(object.getName(), object);
+        }
+
+        return map.build();
+    }
+
+    public static <T extends Object> boolean hasAnyDynamic(Iterable<ValueResolver<T>> resolvers)
+    {
+        for (ValueResolver resolver : resolvers)
+        {
+            if (resolver.isDynamic())
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static boolean isSimpleExpression(String expression, TemplateParser parser)
+    {
+        TemplateParser.PatternInfo style = parser.getStyle();
+        return expression.startsWith(style.getPrefix()) && expression.endsWith(style.getSuffix());
+    }
+
+    public static boolean containsExpression(String expression, TemplateParser parser)
+    {
+        return parser.isContainsTemplate(expression);
+    }
+
+    public static boolean isExpression(Object value, TemplateParser parser)
+    {
+        if (value instanceof String)
+        {
+            String maybeExpression = (String) value;
+            return isSimpleExpression(maybeExpression, parser) || containsExpression(maybeExpression, parser);
+        }
+
+        return false;
+    }
+
+    public static <T extends Described> List<T> alphaSortDescribedList(List<T> list)
+    {
+        if (CollectionUtils.isEmpty(list))
+        {
+            return list;
+        }
+
+        Collections.sort(list, new DescribedComparator());
+        return list;
+    }
+
+    public static <T> ConfigurationInstanceProvider<T> createConfigurationInstanceProvider(String name,
```

Comment:
not for now. This part will undergo deep changes so I prefer to keep it as something to consider when that happens